specifyweb/specify/migrations/0002_geo.py

```diff
@@ -0,0 +1,267 @@
+                ('yesno2', models.BooleanField(blank=True, db_column='YesNo2', null=True)),
+                ('yesno3', models.BooleanField(blank=True, db_column='YesNo3', null=True)),
+                ('parentcog', models.ForeignKey(db_column='ParentCOGID', on_delete=django.db.models.deletion.CASCADE, related_name='parentcojos', to='specify.collectionobjectgroup')),
+                ('childcog', models.OneToOneField(db_column='ChildCOGID', null=True, on_delete=django.db.models.deletion.CASCADE, related_name='cojo', to='specify.collectionobjectgroup')),
+                ('childco', models.OneToOneField(db_column='ChildCOID', null=True, on_delete=django.db.models.deletion.CASCADE, related_name='cojo', to='specify.collectionobject')),
+            ],
+            options={
+                'db_table': 'collectionobjectgroupjoin',
+                'ordering': (),
+                'unique_together': (('parentcog', 'childco'),),
+            },
+        ),
+        migrations.AddField(
+            model_name='geographytreedef',
+            name='discipline',
+            field=models.ForeignKey(db_column='DisciplineID', default=None, null=True, on_delete=protect_with_blockers, related_name='geographytreedefs', to='specify.discipline'),
+            preserve_default=False,
+        ),
+        migrations.AddField(
+            model_name='geologictimeperiodtreedef',
+            name='discipline',
+            field=models.ForeignKey(db_column='DisciplineID', default=None, null=True, on_delete=protect_with_blockers, related_name='geologictimeperiodtreedefs', to='specify.discipline'),
+            preserve_default=False,
+        ),
+        migrations.AddField(
+            model_name='lithostrattreedef',
+            name='discipline',
+            field=models.ForeignKey(db_column='DisciplineID', default=None, null=True, on_delete=protect_with_blockers, related_name='lithostratstreedefs', to='specify.discipline'),
+            preserve_default=False,
+        ),
+        migrations.AddField(
+            model_name='storagetreedef',
+            name='institution',
+            field=models.ForeignKey(db_column='InstitutionID', default=None, null=True, on_delete=protect_with_blockers, related_name='storagetreedefs', to='specify.institution'),
+            preserve_default=False,
+        ),
+        migrations.AddField(
+            model_name='taxontreedef',
+            name='discipline',
+            field=models.ForeignKey(db_column='DisciplineID', default=None, null=True, on_delete=protect_with_blockers, related_name='taxontreedefs', to='specify.discipline'),
+            preserve_default=False,
+        ),
+        migrations.AlterField(
+            model_name='discipline',
+            name='taxontreedef',
+            field=models.OneToOneField(db_column='TaxonTreeDefID', null=True, on_delete=protect_with_blockers, related_name='defaultdiscipline', to='specify.taxontreedef'),
```

Comment:
I'll recommend against making it a OneToOne. From https://github.com/specify/specify7/pull/4977, you should remember that the issue was that some insititutions share treedef. That is, multiple disciplines sharing same treedef. Not only do you block it, but also in the code where you assign disciplines to treedefs, these treedef will be assigned the discipline arbitrarily. 

At least make this not OneToOne AND think over if there is a downside to have picked discipline arbitrarily. (also, think why I mention it'll be picked arbitarily)

Reply:
Yep, realized this too and am working on a fix (for an unrelated problem with the Datamodel). 

I think it should absolutely be `many-to-one`, like with the other TreeDefs. 

(The unrelated problem was that the Institution relationship on StorageTreeDef was commented out).